certification/internal/engine/engine.go

```diff
@@ -279,3 +298,126 @@ func untar(dst string, r io.Reader) error {
+
+	digest, _ := img.Digest()
+	manifest, _ := img.Manifest()
+	rawConfig, _ := img.RawConfigFile()
+	size, _ := img.Size()
+	layers, _ := img.Layers()
+	labels := convertLabels(config.Config.Labels)
+	layerSize := make([]pyxis.Layer, 10)
+
+	certImage := pyxis.CertImage{
+		DockerImageDigest: digest.String(),
+		DockerImageID:     manifest.Config.Digest.String(),
+		ImageID:           digest.String(),
+		ParsedData: &pyxis.ParsedData{
+			Architecture:  config.Architecture,
+			Command:       strings.Join(config.Config.Cmd, " "),
+			Created:       config.Created.String(),
+			DockerVersion: config.DockerVersion,
+			ImageID:       digest.String(),
+			Labels:        labels,
+			OS:            config.OS,
+			Size:          size,
+			//TODO: figure out how to get/convert this data from crane
+			UncompressedLayerSizes: layerSize,
+		},
+		RawConfig:         string(rawConfig),
+		SumLayerSizeBytes: sumLayerSizeBytes(layers),
+		//TODO: figure out how to get/convert this data from crane
+		UncompressedTopLayerId: "placeholder",
+	}
+
+	// calling MarshalIndent so the json file written to disk is human-readable when opened
+	certImageJson, _ := json.MarshalIndent(certImage, "", "    ")
+
+	fileName, err := artifacts.WriteFile(viper.GetString("rpm_manifest_file_name"), string(certImageJson))
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrSaveFileFailed, err)
+	}
+
+	log.Tracef("image config written to disk: %s", fileName)
+
+	return nil
+}
+
+func writeRPMManifest(containerFSPath string) error {
+	db, err := rpmdb.Open(filepath.Join(containerFSPath, "var", "lib", "rpm", "Packages"))
+	if err != nil {
+		return err
+	}
+
+	pkgList, err := db.ListPackages()
+	if err != nil {
+		return err
+	}
+
+	// covert rpm struct to pxyis struct
+	rpms := make([]pyxis.RPM, len(pkgList))
+	for _, packageInfo := range pkgList {
+
+		bgName := packageInfo.SourceRpm[0:strings.LastIndex(strings.SplitAfter(packageInfo.SourceRpm, ".")[0], "-")]
+		endChop := strings.TrimPrefix(strings.TrimSuffix(regexp.MustCompile("(-[0-9].*)").FindString(packageInfo.SourceRpm), ".rpm"), "-")
+
+		rpm := pyxis.RPM{
+			Architecture: packageInfo.Arch,
+			Gpg:          "placeholder", //TODO: change this once GPG is implemented in go-rpm library
+			Name:         packageInfo.Name,
+			Nvra:         fmt.Sprintf("%s-%s-%s.%s", packageInfo.Name, packageInfo.Version, packageInfo.Release, packageInfo.Arch),
+			Release:      packageInfo.Release,
+			SrpmName:     bgName,
+			SrpmNevra:    fmt.Sprintf("%s-%d:%s", bgName, packageInfo.Epoch, endChop),
+			Summary:      packageInfo.Summary,
+			Version:      packageInfo.Version,
+		}
+
+		rpms = append(rpms, rpm)
+	}
+
+	rpmManifest := pyxis.RPMManifest{
+		RPMS: rpms,
+	}
+
+	// calling MarshalIndent so the json file written to disk is human-readable when opened
+	rpmManifestJson, _ := json.MarshalIndent(rpmManifest, "", "    ")
+	fileName, err := artifacts.WriteFile(viper.GetString("rpm_manifest_file_name"), string(rpmManifestJson))
```

Comment:
It would be good to ensure `len(viper.GetString("rpm_manifest_file_name") >0` before writing to the file.

Reply:
This is coming from `default`, but I can add a check.

Reply:
And it can't be overwritten with an envvar, right?

Reply:
@acornett21 if the value is required, we better do a check. There is a chance that the value is overwritten by empty string via env var (not sure if viper is smart enough to disregard the empty value and use the default in this case)

Reply:
@bcrochet Is there a way to check that to be certain?

Reply:
As discussed, these are not expected to be changed by the end user, and finding a package to place these string constants in is the better approach.

Reply:
I created a new `defaults.go` file in the `certification` package. This doesn't create a circular dependency that I was trying to avoid by using viper, or the other `default.go` files in say `cmd`.